core/src/main/scala/io/snappydata/impl/SparkShellRDDHelper.scala

```diff
@@ -56,11 +56,14 @@ final class SparkShellRDDHelper {
     DriverRegistry.register(Constant.JDBC_CLIENT_DRIVER)
     val resolvedName = StoreUtils.lookupName(tableName, conn.getSchema)
     val par = split.index
-    val statement = conn.createStatement()
 
-    if (!useLocatorURL)
-      statement.execute(s"call sys.SET_BUCKETS_FOR_LOCAL_EXECUTION('$resolvedName', $par)")
+    if (!useLocatorURL) {
```

Comment:
Intention was to have separate statement for each query to make it look neat. But I now realize that it is not supposed to be efficient.
I'll revert this specific change.